compiler/sem/injectdestructors.nim

```diff
@@ -408,13 +408,7 @@ func initEntityDict(tree: MirTree, owner: PSym): EntityDict =
 
       if t != nil and hasDestructor(t):
         let re = toName(entity)
-        # XXX: because of an issue with semantic analysis (see
-        #      ``tests/arc/tstrformat.nim``) it can happen that the same entity
-        #      is defined multiple times. In order to not silently generate
-        #      incorrect code, ``doAssert`` is misused to report an internal
-        #      compiler error here. Using the proper facility would require
-        #      access to a ``ConfigRef`` (which we neither do nor should have)
-        doAssert re notin result, "entity appears in a 'def' multiple times"
+        assert re notin result, "entity appears in a 'def' multiple times"
```

Comment:
I would consider leaving it as a `doAssert` until we can get back to `release` builds in CI.

Reply:
I changed it back to `doAssert`.